opentelemetry-stdout/src/logs/exporter.rs

```diff
@@ -44,108 +30,77 @@ impl fmt::Debug for LogExporter {
+            return Err("exporter is shut down".into());
         } else {
-            Err("exporter is shut down".into())
+            if self.resource_emitted {
+                print_logs(batch);
+            } else {
+                self.resource_emitted = true;
+                println!("Resource");
+                if self.resource.schema_url().is_some() {
+                    println!("\t Resource SchemaUrl: {:?}", self.resource.schema_url());
+                }
+
+                self.resource.iter().for_each(|(k, v)| {
+                    println!("\t ->  {}={:?}", k, v);
+                });
+
+                print_logs(batch);
+            }
+
+            Ok(())
         }
     }
 
     fn shutdown(&mut self) {
-        self.writer.take();
+        self.is_shutdown.store(true, atomic::Ordering::SeqCst);
     }
 
     fn set_resource(&mut self, res: &opentelemetry_sdk::Resource) {
         self.resource = res.clone();
     }
 }
 
-/// Stdout exporter's error
-#[derive(thiserror::Error, Debug)]
-#[error(transparent)]
-struct Error(#[from] std::io::Error);
-
-impl ExportError for Error {
-    fn exporter_name(&self) -> &'static str {
-        "stdout"
-    }
-}
-
-/// Configuration for the stdout log exporter
-#[derive(Default)]
-pub struct LogExporterBuilder {
-    writer: Option<Box<dyn Write + Send + Sync>>,
-    encoder: Option<Encoder>,
-}
-
-impl fmt::Debug for LogExporterBuilder {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.write_str("LogExporterBuilder")
-    }
-}
-
-impl LogExporterBuilder {
-    /// Set the writer that the exporter will write to
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use opentelemetry_stdout::LogExporterBuilder;
-    ///
-    /// let buffer = Vec::new(); // Any type that implements `Write`
-    /// let exporter = LogExporterBuilder::default().with_writer(buffer).build();
-    /// ```
-    pub fn with_writer<W>(mut self, writer: W) -> Self
-    where
-        W: Write + Send + Sync + 'static,
-    {
-        self.writer = Some(Box::new(writer));
-        self
-    }
-
-    /// Set the encoder that the exporter will use.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use opentelemetry_stdout::LogExporterBuilder;
-    /// use serde_json;
-    ///
-    /// let exporter = LogExporterBuilder::default()
-    ///     .with_encoder(|writer, data|
-    ///          Ok(serde_json::to_writer_pretty(writer, &data).unwrap()))
-    ///     .build();
-    /// ```
-    pub fn with_encoder<E>(mut self, encoder: E) -> Self
-    where
-        E: Fn(&mut dyn Write, crate::logs::transform::LogData) -> LogResult<()>
-            + Send
-            + Sync
-            + 'static,
-    {
-        self.encoder = Some(Box::new(encoder));
-        self
-    }
+fn print_logs(batch: Vec<Cow<'_, LogData>>) {
+    for (i, log) in batch.into_iter().enumerate() {
+        println!("Log #{}", i);
+        if let Some(event_name) = &log.record.event_name {
```

Comment:
Do we not have to print information about `LogData.InstrumentationLibrary`?

Reply:
yes and no! Its hard (perf wise) to populate InstrumentationLibrary in hot path, so Log appenders are populating it in target.... At somepoint, the target has to be mapped back to Instrumentation Library. Need to follow up on that later.